worker/firewaller/firewaller.go

```diff
@@ -804,8 +810,43 @@ func (fw *Firewaller) updateForRemoteRelationIngress(appTag names.ApplicationTag
 			continue
 		}
 		for _, cidr := range data.networks.Values() {
-			cidrs.Add(cidr)
+			newCidrs.Add(cidr)
+		}
+	}
+	// If we have too many CIDRs to create a rule for, consolidate.
+	// If a firewall rule with a whitelist of CIDRs has been set up,
+	// use that, else open to the world.
+	if newCidrs.Size() > maxAllowedCIDRS {
+		// First, try and merge the cidrs.
+		merged, err := cidrman.MergeCIDRs(newCidrs.Values())
+		if err != nil {
+			return errors.Trace(err)
 		}
+		newCidrs = set.NewStrings(merged...)
+	}
+
+	// If there's still too many after merging, look for any firewall whitelist.
```

Comment:
I don't follow the logic of just using the existing whitelist work? It might not include the new CIDRs...?

Reply:
If any requested ingress cidrs are not in the whitelist, assuming a whitelist has been defined, the request for ingress is rejected. So falling back to the whitelist cides for the firewall rule for the cloud is fine - it just reinforces the existing restriction.